app/api/v1/user/index.js

```diff
@@ -163,4 +86,81 @@ router.route('/bonus')
     }).catch((err) => next(err));
   });
 
+router.route('/:uuid?')
+
+  /**
+   * Gets user profile for current user.
+   */
+  .get((req, res, next) => {
+    if (!req.params.uuid || req.params.uuid === req.user.uuid) {
+      res.json({ error: null, user: req.user.getUserProfile() });
+      return;
+    }
+    User.findByUUID(req.params.uuid).then((user) => {
+      res.json({ error: null, user: user.getPublicProfile() });
+    }).catch(next);
+  })
+
+  /**
+   * Updates user information, given a partial 'user' object in the request body, and returns
+   * the user's updated public profile. Fields that may be updated: firstName, lastName, major,
+   * graduationYear. To update the password, the 'user' object must include matching 'newPassword'
+   * and 'confirmPassword' fields.
+   */
+  .patch((req, res, next) => {
+    if (!req.body.user) return next(new error.BadRequest('User object must be provided'));
+
+    // constructs new, sanitized user object with updated fields
+    const updatedUser = {};
+    // for each of the required user fields, validate input and verify the value has changed
+    const validateStringField = (updatedField) => updatedField && updatedField.length > 0;
+    if (validateStringField(req.body.user.firstName) && req.body.user.firstName !== req.user.firstName) {
+      updatedUser.firstName = req.body.user.firstName;
+    }
+    if (validateStringField(req.body.user.lastName) && req.body.user.lastName !== req.user.lastName) {
+      updatedUser.lastName = req.body.user.lastName;
+    }
+    if (validateStringField(req.body.user.major) && req.body.user.major !== req.user.major) {
+      updatedUser.major = req.body.user.major;
+    }
+    if (req.body.user.graduationYear && !Number.isNaN(parseInt(req.body.user.graduationYear, 10))) {
```

Comment:
Should we change the password requirements with some [regex checking](https://stackoverflow.com/questions/19605150/regex-for-password-must-contain-at-least-eight-characters-at-least-one-number-a)?

Reply:
Thanks for the link; definitely something we can look into in a future PR. I don't particularly care about enforcing password requirements like "must have a special character" but if the rest of the team thinks it's a good idea, I'm all for it.